os_is_not_exist_matcher.go

```diff
@@ -0,0 +1,32 @@
+package gomegamatchers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/onsi/gomega/types"
+)
+
+func BeAnOsIsNotExistError() types.GomegaMatcher {
```

Comment:
Haven't written a gomega matcher in a long time.  is there a reason you have a function returning a matcher?  Why not just have `BeAnOsIsNotExistError` have type `types.GomegaMatcher` (make it a struct that implements the interface), rather than type `func() types.GomegaMatcher` where it's a function which returns a struct which implements the interface?